cpp/include/tensorrt_llm/executor/tensor.h

```diff
@@ -45,6 +50,12 @@ std::shared_ptr<runtime::ITensor> const& toITensor(Tensor const& tensor);
 Tensor ofITensor(std::shared_ptr<runtime::ITensor> tensor);
 using DimType64 = int64_t;
 
+/// @brief Convert executor::DataType to nvinfer1::DataType
+/// @param dataType The executor data type to convert
+/// @return The corresponding nvinfer1::DataType
+/// @throws If dataType is kUNKNOWN or unsupported
+nvinfer1::DataType toTrtDataType(DataType dataType);
```

Comment:
Can this be in another header that would avoid the fwd declaration?

Reply:
I'm avoiding pulling TensorRT headers into this widely used public header